### ISSUE TITLE
Winecfg and regedit windows non-functional

### ISSUE THREAD

**Issue by wundrweapon** [Winetricks/winetricks#1918]
Running winetricks either directly or through protontricks, the winecfg and regedit windows for multiple games do not capture the mouse. These windows used to work just fine. Keyboard controls also do not work. Without one or both, I am unable to make easy changes to these settings.  
Additionally, regedit's window is only half visible. When floating, the right half is just black; when tiling, the window doesn't rescale and simply cuts off.  
Curiously, in the protontricks case, wine spits out numerous "unhandled window class" error messages, including `L"Button"` and `L"SysListView32"`.

I am using 64-bit GNU/Linux with Wayland (my compositor is Sway, so xwayland is handled automatically).

**Comment by austin987** [comment]
> Running winetricks either directly or through protontricks, the winecfg and regedit windows for multiple games do not capture the mouse.

This is almost certainly not a winetricks issue.

> These windows used to work just fine.

What changed since then? I.e., did you upgrade wine/wayland?

> Keyboard controls also do not work. Without one or both, I am unable to make easy changes to these settings. Additionally, regedit's window is only half visible. When floating, the right half is just black; when tiling, the window doesn't rescale and simply cuts off. Curiously, in the protontricks case, wine spits out numerous "unhandled window class" error messages, including `L"Button"` and `L"SysListView32"`.

Does it work if you don't use winetricks to run winecfg/regedit? I.e., if you just run 'wine regedit', does it occur?

**Comment by wundrweapon** [comment]
> if you just run 'wine regedit', does it occur?

To my surprise… no, regedit and winecfg work just fine if I launch `wine` or `wine64` directly.

> What changed since then?

Probably a lot -- I update my system frequently and I don't have to use winecfg or regedit very often. No doubt Proton and my distro's wine package have both updated.

**Comment by austin987** [comment]
> > if you just run 'wine regedit', does it occur?
> 
> To my surprise… no, regedit and winecfg work just fine if I launch `wine` or `wine64` directly.

Well, that's interesting, since all winetricks is doing is running '${WINE} regedit'. Are you testing in a different prefix, maybe? Or maybe wayland does something different when running from the console? What does `winetricks list-installed` say?

> > What changed since then?
> 
> Probably a lot -- I update my system frequently and I don't have to use winecfg or regedit very often. No doubt Proton and my distro's wine package have both updated.

I'd suggest checking what changed on your system. The relevant code in winetricks hasn't changed in years, and I can't reproduce this under X11. I's almost certainly not a winetricks issue (but rather, wine or wayland).

**Comment by wundrweapon** [comment]
> What does `winetricks list-installed` say?

On its own, just "winxp" twice. In different Steam games' compatdata/id/pfx directories (using `WINEPREFIX=/path/to/pfx winetricks list-installed`), things like "grabfullscreen=y" (which makes sense as I would have done `protontricks gameid grabfullscreen=y`)

> I'd suggest checking what changed on your system.

Well… since opening this issue I've updated my system again (wthiout rebooting) and just to make sure I'm not crazy, I tried again.  
They work. They both work, even with protontricks and winetricks. I'll reopen this if I can reproduce it again.
